lib/sqlalchemy/orm/session.py

```diff
@@ -3580,6 +3580,129 @@ def get(
             bind_arguments=bind_arguments,
         )
 
+    def get_one(
+        self,
+        entity: _EntityBindKey[_O],
+        ident: _PKIdentityArgument,
+        *,
+        options: Optional[Sequence[ORMOption]] = None,
+        populate_existing: bool = False,
+        with_for_update: ForUpdateParameter = None,
+        identity_token: Optional[Any] = None,
+        execution_options: OrmExecuteOptionsParameter = util.EMPTY_DICT,
+        bind_arguments: Optional[_BindArguments] = None,
+    ) -> _O:
+        """Return exactly one instance based on the given primary key identifier,
+           or raise an exception.
+
+           Raises ``sqlalchemy.orm.exc.NoResultFound`` if the query selects
+           no rows.
+        E.g.::
+
+            my_user = session.get_one(User, 5)
+
+            some_object = session.get_one(VersionedFoo, (5, 10))
+
+            some_object = session.get_one(
+                VersionedFoo,
+                {"id": 5, "version_id": 10}
+            )
+
+        :meth:`_orm.Session.get_one` is special in that it provides direct
+        access to the identity map of the :class:`.Session`.
+        If the given primary key identifier is present
+        in the local identity map, the object is returned
+        directly from this collection and no SQL is emitted,
+        unless the object has been marked fully expired.
+        If not present,
```

Comment:
**mike bayer** (zzzeek) wrote:

"impl" is not the variable name i would use here, it usually means "implementation", like when you have an abstract  / proxy of some kind and "impl" is the implementation.  it's used in attributes.py.   so here it should be replaced with the more common term we have for an instance of a user-mapped class which is "instance".

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4872

Reply:
Variable name updated, thanks for the review @zzzeek!